### ISSUE TITLE
Comments lost from multi-line expressions

### ISSUE THREAD

**Issue by SteveGilham** [fsprojects/fantomas#2525]
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKBtAPAQwM68QFsAjAGwE80BlAVwAc6AnRfAWRd2wHNEAKAHUgBxRADt42RoUQA6AEo1SLGQAVEjAGawp2UdESCANP0b9REC5avWbYQQEEAbrACW8OYjqIALi9%2BPEAGFsUlJcABVYFUZYL0ZfFgUlIxMzW3SMsCo4LzAAXjtIaRJ1IwgAenKwdm8ACwQAEUQNF1E-F1hRVPNM3stwyR5vfML7Um9A2ADGeUQ4RiQZgElRXG9dfXKl8uDQ8MZsaABrFBRWtY2%2BAEoU0x6%2BvoApGjWXFuhsX06RwQAhcm8cwQiDASHG2DAnQogiuAD4ALrdXhXSCGcBQAjeXyiLi4VBgdDoiCgayQAAk3nIXnxkFa3lRRIs5McIRoyDQhNsAAY0bZICskKJvFQXAAvZC8mwAFkZEHhjIAvpLibKoBSqeyoHSGaSySzSGz8ZybABGZVWSCsbAADwAMq1ELaxFw6jrTQAmLmq%2BWWJWMkkW9XUtCQMTwADyGntokQVEpSjdlmZrM1xus7vNSYAouJI9GJaqIJBSBpIN7FeaA0mg5rIMRYLBSImmXqU0bC2AAMyZpkrAjxABirRCADlEAB3UgO5uWbyMNnl32V1XkynBqD1xsztX6w0cjtSntFqh0Q6IX7NbSIFSSbDSQGMbcQOcL6w%2Bix%2ByxVltr2ubps9smBqph2ACsR5QCeZ4XlozC2rA47qB8BArM4HxfKIT5gC%2BiCLh%2By66r%2BNL-tuQF7gSHYAGwQZAUH6DBV4AKoMEheCIKhsDoR0mEQRoIQEHhECfhY35FjWxENgBK6tsB7a2AA7DRdHnpezCBKQeC4JMqwvtA3jaFhfFhLhb4Vv60lESGJGATJ5FplYAAcSmnvRqmIOwJSPrx-EmVY75CQRgaWRukmkbZIG2AAnM50FuZMpCdIZPmCWAwkqoRGoSVuNm7hFpo8h2tEuYg9gaA%2BkyEIQ2BYThKVpWAolqsFdahTlbb7rYJpmoVykMcwVBEC4cAJTxHZGQJplLuZGXri12XSblckejF%2BilQ%2BA2EENjaJRBtWTfh01BZlVmtQt7UUZ13Y9cV9gxDQ4hNFOm0PjV86%2BZY-mpYF1bNdqbWyR1pqHoVVrWksGjhLUYhUPU8QAOpuK6EFevtAWHT9x1akKYWLYD1gmuBIM2uDkNiFmxkw9o3gI-ASNUSjflmV%2BFmY7S2P-XZHYmtRRNg6ILTWuGcSfNoWbWkwHDcVhDkMx9TMiSzs1-WdAMXaaim8x48zwDTdO2FKssWJ99WNaurPK7quNq-jTma0CCwjjQnmRksgKEHiEEmnV30-qzhCKL4U4xgO2hVVi6jhJlHN5fj0WFVr2jwKwAcuEHiAhzo4deYV0C1LeenqAA%2BuOiO1GWqNfejvtK%2BzKuc7Yno0aDt0HOQ4aMPaay62XEEy97Vdib9teW%2Bd9mWO63V8s3jCt%2B3nfeI7zsaK7RAe1z-fMzNtb%2B%2BMqcOhnYcPpH1LR0t6YZoVLfYG3HcuGsye72nB%2BfC9NG5-nD7F6X5eM1Nm9HTXekp88ZWHdFdKeNoABqKZfitHgK0Lg3dpaGzlPLdKADawW0DFbMeFh3TAwgdaAc909LcVgeIBBSCIIGw3grLeNIsHVhwR2d0hNCENFgN4EQ3gxYS3wNxKhHY%2B4VxNorP8p0R6q1wRAd0PMp4pzTr8BKxxfgHGOD4XA4ZRBUDvEERsTtRq2HGu9I2aCGpiKylJSR9cbDug1nyMck4HQXm8IhMQx9EBNBaG0DC9hxAeWIOoNetg9q-wOv-DGs1rJ1xjqA22fIxguC4KIYhegMIimSZ8GgzBIgCjEOsDCSVjK0PQZE8R81rGxPHnHBJ4x1CiE%2BC4AI8EcQBPUF41o7ROjBJsMYkp5j6EnQqdg0eHZOwFXkbvfMto7zEAkOpWAuAEGOLTkUiaYS0YROruUqxIypFjMnjYSAvD1AuGkEKEIABpRAng8lChWKojYPcxrJRET7QerNomVLPlYTsF8%2BRKN0EcfMuBbqwHuvAMcaxEBJwUQ6XhzB%2BHdNeq%2BDZlctkfKiRIvZNjrCdnAUc34kg%2BqeLvtARgZzhyAngIxNonQmjQA0gcQp3lilvIHk1T52KmGjNsJ2AhRyTkUvOesUgcY7prHnHQOM5AEysvWXLP%2BdCMEMOHjiqpFhOxsKOdczwoNF6BMYJGQFohgUOh6fjLkKCwDG3eZyrFwyeX7L5XIo54qhreFYMCNZJjUG%2BnRPCSUtJcADiWfiPpCogA)

#### Code

```fsharp
[<assembly: SuppressMessage("Gendarme.Rules.Performance",
                            "AvoidRepetitiveCallsToPropertiesRule",
                            Scope = "member",  // MethodDefinition
                            Target = "AltCover.Recorder.Instance/I/CallTrack::instance()",
                            Justification = "Bytecode delta only")>]
()
```
            

#### Result

```fsharp
[<assembly: SuppressMessage("Gendarme.Rules.Performance",
                            "AvoidRepetitiveCallsToPropertiesRule",
                            Scope = "member",
                            Target = "AltCover.Recorder.Instance/I/CallTrack::instance()",
                            Justification = "Bytecode delta only")>]
()

```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#bug-reports).

#### Extra information

- [ ] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-09-20T21:08:34Z - 11c2a862e2b3828627c19ea339f2db48d556e268

Default Fantomas configuration

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

**Comment by nojaf** [comment]
Hello, we have some [documentation](https://fsprojects.github.io/fantomas/docs/contributors/The%20Missing%20Comment.html) on how this issue can be addressed.

Are you interested in submitting a PR?

**Comment by SteveGilham** [comment]
Lab notes:

So, in this case we have trivia, but no node.  The naive approach would be to aim to get the line comment to end up on the same line as the original immediately preceding token, though possibly with other tokens now between them.  If necessary, multiple line comment tokens would be added, ordered by line of origin.

**Comment by nojaf** [comment]
You basically want to pick the best possible node in the tree:
![image](https://user-images.githubusercontent.com/2621499/192276953-31f8ea3c-13ea-4edc-b2e4-2fb2dbc75c65.png)

I would go for `SynExpr_App (3, 28 - 3, 44)`.

Notice that `SynAttribute_ (1, 12 - 1, 27)` only covers the word `SuppressMessage`.
Because of this incorrect range, the children of `SynAttribute_` are never considered.

We could use a different (artificial) range when we create the node in `ASTTransformer` and when we print its trivia in `CodePrinter`.

In the meantime, I've created https://github.com/dotnet/fsharp/issues/13983 to follow up on this.

**Comment by SteveGilham** [comment]
PR #2530 proposed as addressing the issue

**Comment by nojaf** [comment]
This is available in the 5.1 series.
